File: examples_py/basics/connect.py
#!/usr/bin/env python3
"""basics/connect - connection lifecycle: Initialize / IsConnected / Shutdown.

Usage:
    python examples_py/basics/connect.py [--robot-ip <ip>]

    - Pass --robot-ip <ip>, such as 192.168.1.100, to connect to a remote robot.
    - Omit --robot-ip for local simulation.
"""

import argparse
import sys

from rcore_sdk import Robot


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "--robot-ip",
        default="",
        help="Robot IP (omit for local simulation)",
    )
    args = parser.parse_args()
    robot_ip = args.robot_ip

    robot = Robot()
    if not robot.Initialize(robot_ip):
        print(f'Initialize failed (ip="{robot_ip}")', file=sys.stderr)
        return 1

    print(f"Initialize succeeded, IsConnected = {robot.IsConnected()}")

    robot.Shutdown()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/basics/connect.cpp
// basics/connect - connection lifecycle: Initialize / IsConnected / Shutdown
//
// Usage: ./basics_connect [--robot-ip <ip>]
//   - Pass --robot-ip <ip>, such as 192.168.1.100, to connect to a remote robot.
//   - Omit --robot-ip for local simulation.

#include "sdk/robot.hpp"

#include <cstdio>
#include <string>

int main(int argc, char **argv)
{
    std::string robot_ip;
    for (int i = 1; i < argc; ++i)
    {
        const std::string arg = argv[i];
        if (arg == "--robot-ip" && i + 1 < argc)
        {
            robot_ip = argv[++i];
        }
        else
        {
            std::fprintf(stderr, "Usage: %s [--robot-ip <ip>]\n", argv[0]);
            return 1;
        }
    }

    rcore::sdk::Robot robot;
    if (!robot.Initialize(robot_ip))
    {
        std::fprintf(stderr, "Initialize failed (ip=\"%s\")\n", robot_ip.c_str());
        return 1;
    }

    std::printf("Initialize succeeded, IsConnected = %s\n",
                robot.IsConnected() ? "true" : "false");

    robot.Shutdown();
    return 0;
}
